src/components/TeachersPage/TeachersItem/TeachersItem.tsx

```diff
@@ -78,14 +83,23 @@ const TeacherItem: React.FC<TeacherItemProps> = ({
             Price / 1 hour:{' '}
             <span style={{ color: 'green' }}>{teacher.price_per_hour}$</span>
           </p>
-          {isUserLoggedIn ? (
-            <div style={{ cursor: 'pointer' }}>
-              <NonFavoriteIcon onClick={handleFavoriteClick} color="#8a8a89" />
-              <FavoriteIcon onClick={handleFavoriteClick} color="red" />
-            </div>
-          ) : (
-            ''
-          )}
+          <div style={{ cursor: 'pointer' }}>
+            {isUserLoggedIn ? (
+              isFavorite ? (
+                <button onClick={handleFavoriteClick} className={styles.btnAdd}>
+                  <FavoriteIcon />
+                </button>
+              ) : (
+                <button onClick={handleFavoriteClick} className={styles.btnAdd}>
+                  <NonFavoriteIcon color="#8a8a89" className={styles.favIcon} />
+                </button>
+              )
+            ) : (
+              <button onClick={handleToast} className={styles.btnAdd}>
+                <NonFavoriteIcon color="#8a8a89" className={styles.favIcon} />
+              </button>
+            )}
+          </div>
```

Comment:
If u didnt logged in, while u clicking on favorite button, u need to show popup that use need to authorize
hard logic, change it